test/fzf.vader

```diff
@@ -147,6 +147,23 @@ Execute (fzf#wrap):
   let opts = fzf#wrap({})
   Assert opts.options =~ '^--color=fg:'
 
+Execute (fzf#shellesc with sh):
+  AssertEqual '""', fzf#shellesc('', 'sh')
+  AssertEqual '"\"\""', fzf#shellesc('""', 'sh')
+  AssertEqual '"foobar>"', fzf#shellesc('foobar>', 'sh')
+  AssertEqual '"\\""', fzf#shellesc('\"', 'sh')
```

Comment:
Yeah, actually this was not properly discussed when we added support for list typed options. Currently `call fzf#run({'options': ['--prompt', '$USER']})` gives literal `$USER` due to the use of builtin `shellescape`, and I think it makes sense as one can manually expand the var when needed: `call fzf#run({'options': ['--prompt', expand('$USER')]})`.

I briefly looked at the other places where `s:shellesc` is used and it appeared that not expanding env vars anymore will not break anything.